create/cloudvm.go

```diff
@@ -0,0 +1,141 @@
+	PowerState          string            `default:"on" help:"PowerState specifies the power state of the cloud VM. A value of On turns the VM on, shutdown sends an ACPI signal to the VM to perform a clean shutdown and off forces the power off immediately."`
+	OS                  string            `default:"" help:"OS which should be used to boot the VM."`
+	BootDiskSize        string            `default:"20Gi" help:"BootDiskSize that will be used to boot the VM from."`
+	Disks               map[string]string `default:"{}" help:"Disks specifies which additional disks to mount to the machine."`
+	PublicKeys          []string          `default:"[]" help:"PublicKeys specifies the SSH Public Keys that can be used to connect to the VM as root. The keys are expected to be in SSH format as defined in RFC4253. Immutable after creation."`
+	PublicKeysFromFile  []string          `short:"public-keys-f" predictor:"file" help:"CloudConfig via file. Has precedence over args. PublicKeys specifies the SSH Public Keys that can be used to connect to the VM as root. The keys are expected to be in SSH format as defined in RFC4253. Immutable after creation."`
+	CloudConfig         string            `default:"" help:"CloudConfig allows to pass custom cloud config data (https://cloudinit.readthedocs.io/en/latest/topics/format.html#cloud-config-data) to the cloud VM. If a CloudConfig is passed, the PublicKey parameter is ignored. Immutable after creation."`
+	CloudConfigFromFile string            `short:"cloud-config-f" predictor:"file" help:"CloudConfig via file. Has precedence over args. CloudConfig allows to pass custom cloud config data (https://cloudinit.readthedocs.io/en/latest/topics/format.html#cloud-config-data) to the cloud VM. If a CloudConfig is passed, the PublicKey parameter is ignored. Immutable after creation."`
+	Wait                bool              `default:"true" help:"Wait until CloudVM is created."`
+	WaitTimeout         time.Duration     `default:"600s" help:"Duration to wait for CloudVM getting ready. Only relevant if --wait is set."`
+}
+
+func (cmd *cloudVMCmd) Run(ctx context.Context, client *api.Client) error {
+	cloudVM, err := cmd.newCloudVM(client.Project)
+	if err != nil {
+		return err
+	}
+
+	c := newCreator(client, cloudVM, strings.ToLower(infrastructure.CloudVirtualMachineKind))
+	ctx, cancel := context.WithTimeout(ctx, cmd.WaitTimeout)
+	defer cancel()
+
+	if err := c.createResource(ctx); err != nil {
+		return err
+	}
+
+	if !cmd.Wait {
+		return nil
+	}
+
+	if err := c.wait(ctx, waitStage{
+		objectList: &infrastructure.CloudVirtualMachineList{},
+		onResult: func(event watch.Event) (bool, error) {
+			if c, ok := event.Object.(*infrastructure.CloudVirtualMachine); ok {
+				return isAvailable(c), nil
+			}
+			return false, nil
+		}},
+	); err != nil {
+		return err
+	}
+
+	fmt.Printf("\n Your Cloud VM %s is now available, you can now connect with ssh %s\n\n", cloudVM.Name, cloudVM.Status.AtProvider.FQDN)
```

Comment:
in my testing, this resulted in an empty fqdn:

```
✓ created cloudvirtualmachine "poetic-songbird" 🏗
 ✓ waiting for cloudvirtualmachine to be ready ⏳
 ✓ cloudvirtualmachine ready 🛫

 Your Cloud VM poetic-songbird is now available, you can now connect with ssh
```

I think this is because we don't get the updated object after the wait. So either do another `Get` here or simply reassign the `cloudVM` in the `onResult`  function where we already get the latest object from the watch.

Additionally, maybe make a line break like `connect with:\n  <ssh command>`, so it's easier to separate and copy the command.

Reply:
fixed, reassigned the object